codegen/args.go

```diff
@@ -0,0 +1,114 @@
+package codegen
+
+import (
+	"fmt"
+	"go/types"
+	"strings"
+
+	"github.com/99designs/gqlgen/codegen/config"
+	"github.com/99designs/gqlgen/codegen/templates"
+	"github.com/pkg/errors"
+	"github.com/vektah/gqlparser/ast"
+)
+
+type ArgSet struct {
+	Args     []*FieldArgument
+	FuncDecl string
+}
+
+type FieldArgument struct {
+	*ast.ArgumentDefinition
+	TypeReference *config.TypeReference
+	VarName       string      // The name of the var in go
+	Object        *Object     // A link back to the parent object
+	Default       interface{} // The default value
+	Directives    []*Directive
+	Value         interface{} // value set in Data
+}
+
+func (f *FieldArgument) Stream() bool {
+	return f.Object != nil && f.Object.Stream
+}
+
+func (b *builder) buildArg(obj *Object, arg *ast.ArgumentDefinition) (*FieldArgument, error) {
+	def := b.Schema.Types[arg.Type.Name()]
+	if !def.IsInputType() {
```

Comment:
Are these sorts of checks more in the realm of parser validation? Should gqlgen always be operating under the assumption that the schema is valid and well-formed?

Reply:
Good point, Will update the parser.

Reply:
Cool, yeah I noticed a few of these around. I think it's a reasonable assumption, and our fixes for codegen panics recently have been to catch issues in parser validation.